src/util.cpp

```diff
@@ -503,39 +505,23 @@ std::vector<std::string> ispc::MatchStrings(const std::string &str, const std::v
     return std::vector<std::string>();
 }
 
-void ispc::GetDirectoryAndFileName(const std::string &currentDirectory, const std::string &relativeName,
-                                   std::string *directory, std::string *filename) {
-#ifdef ISPC_HOST_IS_WINDOWS
-    char path[MAX_PATH];
-    const char *combPath = PathCombine(path, currentDirectory.c_str(), relativeName.c_str());
-    Assert(combPath != nullptr);
-    const char *filenamePtr = PathFindFileName(combPath);
-    *filename = filenamePtr;
-    *directory = std::string(combPath, filenamePtr - combPath);
-#else
-    // We need a fully qualified path.  First, see if the current file name
-    // is fully qualified itself--in that case, the current working
-    // directory isn't needed.
-    // @todo This probably needs to be smarter for Windows...
-    std::string fullPath;
-    if (relativeName[0] == '/')
+std::pair<std::string, std::string> ispc::GetDirectoryAndFileName(const std::string &currentDirectory,
+                                                                  const std::string &relativeName) {
+    llvm::SmallString<256> fullPath;
+
+    if (llvm::sys::path::is_absolute(relativeName)) {
+        // We may actually get not relative path but absolute, then just use it.
         fullPath = relativeName;
-    else {
-        fullPath = g->currentDirectory;
-        if (fullPath[fullPath.size() - 1] != '/')
-            fullPath.push_back('/');
-        fullPath += relativeName;
+    } else {
+        fullPath = currentDirectory;
```

Comment:
Interesting, it used `g->currentDirectory` before. Is it something different?

Reply:
Both use cases have `currentDirector == g->currentDirectory` so nothing changed except the removed inconsistency.